File: tests/fixtures/sample.lua
local PaletteRole = {}
PaletteRole.__index = PaletteRole

function PaletteRole.new(field_value, mutable_count)
  local self = setmetatable({}, PaletteRole)
  self.field_value = field_value
  self.mutable_count = mutable_count
  return self
end

function PaletteRole:format_value(input_value)
  local local_value = string.format("%s:%s:%s", input_value, self.field_value, "BD93F9")
  return local_value:upper()
end

return PaletteRole

File: tests/fixtures/sample.py
class PaletteRole:
    DEFAULT_COLOR = "#BD93F9"

    def __init__(self, field_value: str, mutable_count: int) -> None:
        self.field_value = field_value
        self.mutable_count = mutable_count

    def format_value(self, input_value: str) -> str:
        local_value = f"{input_value}:{self.field_value}:{PaletteRole.DEFAULT_COLOR}"
        return local_value.upper()
